syntax="proto3";

option cc_generic_services = true;

message GetRequest {
    string key = 1;
};

message GetResponse {
    string value = 1;
    bool status = 2;
    bool exist =3;
};

message GetRangeResponse {
    bool exist =1;
    repeated string key = 2;
    repeated string value = 3;
    int32 num = 4;
    bool status = 5;
    
};

message PutRequest {
    string key = 1;
    string value = 2;
};

message PutResponse {
    bool status = 1;
};
message DelRequest {
    string key = 1;
};

message DelResponse {
    bool status = 1;
};

service KVService {
      rpc Get(GetRequest) returns (GetResponse);
      rpc GetRange(GetRequest) returns (GetRangeResponse);
      rpc Put(PutRequest) returns (PutResponse);
      rpc Del(DelRequest) returns (DelResponse);
};